src/test/AludelFactoryIntegration.sol

```diff
@@ -61,17 +64,24 @@ contract AludelFactoryIntegrationTest is DSTest {
     }
 
     uint256 public constant BASE_SHARES_PER_WEI = 1000000;
+    uint256 public constant STAKE_AMOUNT = 1 ether;
+    uint256 public constant REWARD_AMOUNT = 10 ether;
 
     AludelInitializationParams private defaultParams;
 
     function setUp() public {
-        cheats = Hevm(HEVM_ADDRESS);
-        owner = cheats.addr(PRIVATE_KEY);
-
-        recipient = cheats.addr(PRIVATE_KEY + 1);
+        vm = Vm(HEVM_ADDRESS);
+        
         // 100 / 10000 => 1%
         bps = 100;
-        factory = new AludelFactory(recipient, bps);
+
+        UserManager userManager = new UserManager();
```

Comment:
any reason for the UserManager not to store the current index and make it auto-incremental? would it ever be used in a different way?

Reply:
I left that for later